cocos/asset/assets/buffer-asset.ts

```diff
@@ -35,36 +35,38 @@ import { Asset } from './asset';
  */
 @ccclass('cc.BufferAsset')
 export class BufferAsset extends Asset {
-    private _buffer: ArrayBuffer | null = null;
-
     /**
-     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
+     * @zh 缓冲数据的字节视图。
+     * @en Byte view of the buffered data.
      */
-    @override
-    get _nativeAsset () {
-        return this._buffer as ArrayBuffer;
-    }
-    set _nativeAsset (bin: ArrayBufferView | ArrayBuffer) {
-        if (bin instanceof ArrayBuffer) {
-            this._buffer = bin;
-        } else {
-            this._buffer = bin.buffer;
-        }
-    }
+    @serializable
+    public bytes = new Uint8Array();
 
     /**
-     * @zh 获取此资源中的缓冲数据。
-     * @en Get the ArrayBuffer data of this asset.
-     * @returns @en The ArrayBuffer. @zh 缓冲数据。
+     * @zh 首次调用将 **复制** 此时的`this.bytes`，并返回副本的 `ArrayBuffer`；该副本会在后续调用中直接返回。
+     * @en The first invocation on this method will **clone** `this.bytes` and returns `ArrayBuffer` of the copy.
+     * The copy will be directly returned in following invocations.
+     *
+     * @returns @en The `ArrayBuffer`. @zh `ArrayBuffer`。
+     *
+     * @deprecated @zh 自 3.9.0，此方法废弃，调用此方法将带来可观的性能开销；请转而使用 `this.bytes`。
+     * @en Since 3.9.0, this method is deprecated.
+     * Invocation on this method leads to significate cost. Use `this.bytes` instead.
      */
     public buffer () {
-        assertIsNonNullable(this._buffer);
-        return this._buffer;
+        if (!this._bytesLegacy) {
+            this._bytesLegacy = new Uint8Array(this.bytes);
+        }
+        return this._bytesLegacy.buffer;
```

Comment:
This api returns a reference, not an actual copy, so a new temporary Uint8Array doesn't make sense.
Try
```js
const buffer = new ArrayBuffer(16);
const originalUint8Array = new Uint8Array(buffer, 4, 8);
originalUint8Array.set([72, 101, 108, 108, 111, 32, 87, 111]);

// 创建一个新的 ArrayBuffer，大小与原始 ArrayBuffer 中的有效部分相同
const copiedArrayBuffer = new ArrayBuffer(originalUint8Array.byteLength);

// 使用新的 ArrayBuffer 创建一个新的 Uint8Array，并将原始数据复制到新的缓冲区
const copiedUint8Array = new Uint8Array(copiedArrayBuffer);
copiedUint8Array.set(new Uint8Array(originalUint8Array.buffer, originalUint8Array.byteOffset, originalUint8Array.byteLength));

console.log(copiedUint8Array); // Uint8Array(8) [72, 101, 108, 108, 111, 32, 87, 111]
```